services/src/main/java/org/keycloak/services/resources/admin/permissions/ClientPermissions.java

```diff
@@ -292,6 +293,7 @@ public Resource resource(ClientModel client) {
     @Override
     public Map<String, String> getPermissions(ClientModel client) {
         initialize(client);
+        //does it make sense to return the scopes if fine-grained permissions are disabled?
```

Comment:
I don't think it does. The calling code does not call this method if the feature is disabled. See https://github.com/keycloak/keycloak/blob/2b996b12a180c426bbefe5cb44b6fce1af982840/services/src/main/java/org/keycloak/services/resources/admin/ClientResource.java#L660.